src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -1346,6 +1346,48 @@ private Table GetTimeSeriesTable(Transaction tx, CollectionName collection)
             return tx.OpenTable(TimeSeriesSchema, tableName);
         }
 
+        public DynamicJsonArray GetTimeSeriesNamesForDocument(DocumentsOperationContext context, string docId)
+        {
+            var table = new Table(TimeSeriesSchema, context.Transaction.InnerTransaction);
+            var list = new DynamicJsonArray();
+
+            if (table.SeekToLastPrimaryKey(out var reader) == false)
+                return list;
+
+            var holder = new ByteStringStorage
+            {
+                Flags = ByteStringType.Mutable,
+                Length = 0,
+                Ptr = (byte*)0,
+                Size = 0
+            };
+            var slice = new Slice(new ByteString(&holder));
+
+            using (DocumentIdWorker.GetSliceFromId(context, docId, out Slice documentKeyPrefix, SpecialChars.RecordSeparator))
```

Comment:
I don't follow what you are doing here.

Reply:
we need to find all of the time series names for a given document.
for example we have the following segments:
```
            doc/heartbeats/123
            doc/heartbeats/234
            doc/heartbeats/666
            doc/pulse/123
            doc/pulse/54656
```
so we seek backwards, starting from the end.
extracting the last name and use it as a prefix for the next iteration